src/frontend/src/optimizer/plan_node/logical_over_window.rs

```diff
@@ -53,7 +53,41 @@ impl LogicalOverWindow {
         input: PlanRef,
         mut select_exprs: Vec<ExprImpl>,
     ) -> Result<(PlanRef, Vec<ExprImpl>)> {
-        let input_len = input.schema().len();
+        let mut input_proj_builder = ProjectBuilder::default();
+        let _: Vec<_> = input
+            .schema()
+            .fields()
+            .iter()
+            .enumerate()
+            .map(|(idx, x)| input_proj_builder.add_expr(&InputRef::new(idx, x.data_type()).into()))
+            .try_collect()
+            .map_err(|err| ErrorCode::NotImplemented(format!("{err} inside input"), None.into()))?;
```

Comment:
I guess you don't need to add all input columns to the `Project`. You need to add expressions that appear in the `PARTITION BY`, `ORDER BY` and window function arguments to the `Project`, and replace those expressions in these three kinds of places with `InputRef` that references the output of `Project`.

Reply:
I added the expressions involved in order by and partition by directly in addition to the original input column. The redundant columns will be processed in the subsequent column prune step. In other words, I guess there is no need to consider optimization when LogicalOverWindow is created, but only need to ensure that the expression is executed correctly.

Reply:
I was wrong😇. The input columns are indeed required.